pkg/kn/commands/plugin/plugin_verifier_test.go

```diff
@@ -0,0 +1,228 @@
+				return
+			}
+
+			setup(t)
+			defer cleanup(t)
+
+			pluginDir, err := ioutil.TempDir("", "plugin")
+			assert.NilError(t, err)
+			defer os.RemoveAll(pluginDir)
+			pluginPath := filepath.Join(pluginDir, "kn-execution-test")
+			err = ioutil.WriteFile(pluginPath, []byte("#!/bin/sh\ntrue"), 0644)
+			assert.NilError(t, err, "can't create test plugin")
+
+			for _, uid := range getExecTestUids() {
+				for _, gid := range getExecTestGids() {
+					for _, userPerm := range []int{0, UserExecute} {
+						for _, groupPerm := range []int{0, GroupExecute} {
+							for _, otherPerm := range []int{0, OtherExecute} {
+								perm := os.FileMode(userPerm | groupPerm | otherPerm + 0444)
+								assert.NilError(t, prepareFile(pluginPath, uid, gid, perm), "prepare plugin file, uid: %d, gid: %d, perm: %03o", uid, gid, perm)
+
+								eaw := errorsAndWarnings{}
+								eaw = newPluginVerifier(rootCmd).verify(eaw, pluginPath)
+
+								if isExecutable(pluginPath) {
+									assert.Assert(t, len(eaw.warnings) == 0, "executable: perm %03o | uid %d | gid %d | %v", perm, uid, gid, eaw.warnings)
+									assert.Assert(t, len(eaw.errors) == 0)
+								} else {
+									assert.Assert(t, len(eaw.warnings) == 1, "not executable: perm %03o | uid %d | gid %d | %v", perm, uid, gid, eaw.warnings)
+									assert.Assert(t, len(eaw.errors) == 0)
+									assert.Assert(t, util.ContainsAll(eaw.warnings[0], pluginPath))
+								}
+
+							}
+						}
+					}
+				}
+			}
+		})
+
+		t.Run("when kn plugin in path is executable", func(t *testing.T) {
+			setup(t)
+			defer cleanup(t)
+			pluginPath = CreateTestPlugin(t, KnTestPluginName, KnTestPluginScript, FileModeExecutable)
+
+			t.Run("when kn plugin in path shadows another", func(t *testing.T) {
+				var shadowPluginPath = CreateTestPlugin(t, KnTestPluginName, KnTestPluginScript, FileModeExecutable)
+				verifier.seenPlugins[KnTestPluginName] = pluginPath
+				defer DeleteTestPlugin(t, shadowPluginPath)
+
+				t.Run("fails with overshadowed error", func(t *testing.T) {
+					eaw := errorsAndWarnings{}
+					eaw = verifier.verify(eaw, shadowPluginPath)
+					assert.Assert(t, len(eaw.errors) == 0)
+					assert.Assert(t, len(eaw.warnings) == 1)
+					assert.Assert(t, util.ContainsAll(eaw.warnings[0], "shadowed", "ignored"))
+				})
+			})
+		})
+
+		t.Run("when kn plugin in path overwrites existing command", func(t *testing.T) {
+			setup(t)
+			defer cleanup(t)
+			var overwritingPluginPath = CreateTestPlugin(t, "kn-plugin", KnTestPluginScript, FileModeExecutable)
+			defer DeleteTestPlugin(t, overwritingPluginPath)
+
+			t.Run("fails with overwrites error", func(t *testing.T) {
+				eaw := errorsAndWarnings{}
+				eaw = verifier.verify(eaw, overwritingPluginPath)
+				assert.Assert(t, len(eaw.errors) == 1)
+				assert.Assert(t, len(eaw.warnings) == 0)
+				assert.Assert(t, util.ContainsAll(eaw.errors[0], "overwrite", "kn-plugin"))
+			})
+		})
+	})
+}
+
+func isExecutable(plugin string) bool {
```

Comment:
Why are we ignoring output here? Wonder if we should save this for debug purposes?